designs/compatibility/arm64-support.md

```diff
@@ -0,0 +1,150 @@
+
+## Proposal
+
+This proposal involves extending current build and release workflows to produce `arm64` images and validate them through CI/CD. It includes adding `arm64` test runners, ensuring that Helm charts and manifests work on `arm64` clusters, and validating that all dependencies are functional on Arm architecture.
+
+### User Stories
+
+#### Story 1
+
+**As a developer** running a test Kubernetes cluster on my local M-series MacBook, I want to deploy OpenEBS to manage my local storage. Without official `arm64` images, I feel boxed out of developing on top of OpenEBS. Official support ensures I can quickly deploy and maintain OpenEBS on my machine without custom tooling, virtualization, or complicated workarounds.
+
+#### Story 2
+
+**As an operator** managing AWS Graviton-based EKS clusters, I need a storage solution that runs natively and efficiently on `arm64`. With official `arm64` support in OpenEBS, I can deploy OpenEBS alongside my other Arm-native applications, simplifying my environment and reducing operational overhead related to mixed-architecture maintenance.
+
+### Implementation Details/Notes/Constraints
+  
+- **Dependency Checks**:
+  - Verify that all external dependencies (`spdk-rs`, `bitnami-shell`, etc.) provide `arm64` binaries or have acceptable fallbacks.
+
+- **Control-plane Builds**:
+  - Update control-plane builds to support `arm64`.
+
+- **Data-plane build**:
+  - Update data-plane builds to support `arm64`.
+    - LocalPV-HostPath
+    - LocalPV-ZFS
+    - LocalPV-LVM
+    - ReplicatedPV-Mayastor
+  
+- **Testing Strategy**:
+  - Introduce `arm64` test nodes for integration tests.
+  - Run key conformance tests and functional tests on `arm64` platforms to ensure feature parity.
+
+- **Documentation and Examples**:
+  - Update the official OpenEBS docs to include instructions for deploying on `arm64` clusters.
+  - Provide example manifests and Helm chart values showing how to install and verify `arm64` deployments.
+
+### Risks and Mitigations
+
+- **Risk: CI Complexity**: Adding `arm64` pipelines may increase build time and complexity.
+  - *Mitigation*: Use cloud-based ARM runners or Arm-enabled CI platforms, and possibly maintain separate test schedules for `arm64` to reduce load.
+  
+- **Risk: Limited Community Hardware**: Some maintainers and community members might not have immediate access to native `arm64` hardware for debugging.
+  - *Mitigation*: Utilize public cloud credits for temporary Arm instances or rely on well-supported emulation to replicate issues.
+
+- **Risk: External Dependencies**: Some upstream dependencies may lack `arm64` builds.
+  - *Mitigation*: Work closely with dependency maintainers or find equivalents that support `arm64`. Document any known gaps or compatibility issues.
+
+## Graduation Criteria
+
+- **Alpha (Initial Support)**:
+  - `arm64` images are built and published for all primary OpenEBS components.
```

Comment:
Good, I think we need to ensure it's done for all engines.
Though I think we may already be building arm64 for the local engines. @niladrih  @Abhinandan-Purkait ?

Reply:
Yes, we build arm64 images for lvm/zfs/hostpath localpvs.

Reply:
I'll change the language to reflect this.

Reply:
Yeah, the localpv images are built for quite a few archs.